google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -137,6 +165,99 @@ public static void afterClass() throws ExecutionException, InterruptedException
+      if (ex.getStatus().getCode() == Status.Code.NOT_FOUND) {
+        // Create KmsKeyRing
+        String keyRingParent = "parent=projects/" + projectId + "/locations/" + KMS_KEY_RING_LOCATION;
+        CreateKeyRingRequest createKeyRingRequest = CreateKeyRingRequest.newBuilder()
+            .setParent(keyRingParent)
+            .setKeyRingId(KMS_KEY_RING_NAME).build();
+        requestParamsHeader.put(requestParamsKey, keyRingParent);
+        KeyManagementServiceBlockingStub serviceBlockingStubForCreateKeyRing = MetadataUtils
+            .attachHeaders(keyManagementServiceBlockingStub, requestParamsHeader);
+        serviceBlockingStubForCreateKeyRing.createKeyRing(createKeyRingRequest);
+      } else {
+        throw ex;
+      }
+    }
+
+    ServiceAccount serviceAccount = storage.getServiceAccount(projectId);
+    IAMPolicyGrpc.IAMPolicyBlockingStub iamPolicyBlockingStub = IAMPolicyGrpc.newBlockingStub(kmsChannel)
+        .withCallCredentials(MoreCallCredentials.from(credentials));
+    Binding binding = Binding.newBuilder().setRole("roles/cloudkms.cryptoKeyEncrypterDecrypter")
+        .addMembers("serviceAccount:"+serviceAccount.getEmail()).build();
+    com.google.iam.v1.Policy policy = com.google.iam.v1.Policy.newBuilder().addBindings(binding).build();
+    SetIamPolicyRequest setIamPolicyRequest = SetIamPolicyRequest.newBuilder().setResource(kmsKeyRingResourcePath)
+        .setPolicy(policy).build();
+    requestParamsHeader.put(requestParamsKey, "parent="+kmsKeyRingResourcePath);
+    iamPolicyBlockingStub = MetadataUtils.attachHeaders(iamPolicyBlockingStub, requestParamsHeader);
+    iamPolicyBlockingStub.setIamPolicy(setIamPolicyRequest);
+
+    kmsKeyOneResourcePath = CryptoKeyName
+        .of(projectId, KMS_KEY_RING_LOCATION, KMS_KEY_RING_NAME, KMS_KEY_ONE_NAME).toString();
+    try {
+      // Attempt to Get CryptoKey
+      GetCryptoKeyRequest getCryptoKeyOneRequest = GetCryptoKeyRequest.newBuilder()
+          .setName(kmsKeyOneResourcePath).build();
+      keyManagementServiceBlockingStub.getCryptoKey(getCryptoKeyOneRequest);
+    } catch(StatusRuntimeException ex) {
+      if (ex.getStatus().getCode() == Status.Code.NOT_FOUND) {
+        CryptoKey cryptoKey = CryptoKey.newBuilder().setPurpose(CryptoKey.CryptoKeyPurpose.ENCRYPT_DECRYPT).build();
+        CreateCryptoKeyRequest createCryptoKeyOneRequest = CreateCryptoKeyRequest.newBuilder()
+            .setCryptoKeyId(KMS_KEY_ONE_NAME).setParent(kmsKeyRingResourcePath).setCryptoKey(cryptoKey).build();
+        requestParamsHeader.put(requestParamsKey, "parent=" + kmsKeyRingResourcePath);
+        KeyManagementServiceBlockingStub serviceBlockingStubForCreateCryptoKey = MetadataUtils
+            .attachHeaders(keyManagementServiceBlockingStub, requestParamsHeader);
+        serviceBlockingStubForCreateCryptoKey.createCryptoKey(createCryptoKeyOneRequest);
+      } else {
+        throw ex;
+      }
+    }
+
+    kmsKeyTwoResourcePath = CryptoKeyName
```

Comment:
This section and the one above seem to be heavily duplicated - could you extract out a method?